src/dynamic_bridge.cpp

```diff
@@ -554,8 +554,8 @@ int main(int argc, char * argv[])
         return;
       }
 
-      ros1_publishers.clear();
-      ros1_subscribers.clear();
+      std::map<std::string, std::string> current_ros1_publishers;
+      std::map<std::string, std::string> current_ros1_subscribers;
```

Comment:
This PR only fixes the race condition. `active` is imo not a good prefix. Therefore I chose `current`. But I would rather not touch any unrelated code. Any kind of renaming / unification can be done in a separate PR.